### PR TITLE
[WIP] [RfR] Migration for DraftRegistration question.extra

### PR DESCRIPTION
**Pull request by barbour-em**
## Purpose

Having multiple files in a DraftReg requires that the extra field be a list but looking at data on staging most are still dicts. I didn't notice this problem locally which is why this migration wasn't in the original PR. 

## Changes
``` 
'q1' : {                         'q1': {
  'value': 'foo',      ->          'value': 'foo',
  'extra': {}                      'extra': []
}                                 }
'q2': {                          'q2': {
  'value': 'bar',                  'value': 'bar',
  'extra': {                       'extra': [{
    'file': 'data'       ->          'file': 'data'
  }                                 }]
}                                 }
```

## Side effects

Shouldn't be any noticeable side effects other than fixing js errors that might be currently happening on staging. 

## Ticket
No ticket other than [the parent](https://openscience.atlassian.net/browse/OSF-5586)

### CODE REVIEW

**Comment by sloria** [comment]
@samchrisinger Can you review this when you get a chance?

**Comment by sloria** [comment]
Note to whoever merges this--this will need to be merged into `release/0.68.0`.

@barbour-em Be sure to **not** merge develop into this branch.

**Comment by samchrisinger** [comment]
Yes

File: scripts/migration/migrate_registration_extra.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+from website.models import DraftRegistration
+from framework.transactions.context import TokuTransaction
+
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+def migrate_drafts(dry):
+    draft_registrations = DraftRegistration.find()
```
Comment (samchrisinger) [inline scripts/migration/migrate_registration_extra.py]: Is there a reason not to filter this set ahead of time? Like:
https://github.com/CenterForOpenScience/osf.io/blob/847824cf86ddd7403a9b18b1ff3278cd8399ab63/website/prereg/views.py#L26

**Comment by samchrisinger** [comment]
One comment, otherwise lgtm

File: scripts/migration/migrate_registration_extra.py (hunk 1 above)
Comment (barbour-em) [inline scripts/migration/migrate_registration_extra.py]: No, there isn't. Done in fddd5e6

**Comment by barbour-em** [comment]
@nicipfeiffer how's this?
screen shot 2016-04-25 at 8 55 09 am

**Comment by nicipfeiffer** [comment]
@barbour-em Looks good, thank you!

**Comment by samchrisinger** [comment]
@barbour-em do we need to migrate metadata for existing Prereg registrations?

**Comment by barbour-em** [comment]
Thinking about displaying the registration supplement, yes. What should that query look like?

```python
Node.find(
    Q('is_registered', 'eq', True) &
    Q(PREREG_CHALLENGE_METASCHEMA, 'in', 'registered_schema') 
)
```
?

**Comment by samchrisinger** [comment]
^ Maybe add a `&

**Pull request closed without merging**